### PR TITLE
Add addOnceHandler method to EventSource.

### PR DESCRIPTION
**Pull request by avandecreme**
I also would like to add a method to specify the number of times the handler should be called before being removed.
This can also just be a 4th parameter to addHandler.

What do you think?

### CODE REVIEW

File: src/eventsource.js [hunk 1]
```diff
@@ -56,7 +56,22 @@ $.EventSource = function() {
...
+     * @function
+     * @param {String} eventName - Name of event to register.
+     * @param {OpenSeadragon.EventHandler} handler - Function to call when event
+     * is triggered.
+     * @param {Object} [userData=null] - Arbitrary object to be passed unchanged
+     * to the handler.
+     */
+    addOnceHandler: function(eventName, handler, userData) {
+        var self = this;
+        this.addHandler(eventName, function onceHandler(event) {
```
Comment (iangilman) [inline src/eventsource.js]: I think we found that on some browser it didn't like having the function named like that. Instead we had to put the new function in a var and pass that. Unfortunately, I don't remember the exact incident, but maybe it's worth doing that way anyway?

**Comment by iangilman** [comment]
What's the use case for being able to specify a "once" greater than one? I guess I'm not against it, but I'm having a hard time thinking of when you would want it.

**Comment by avandecreme** [comment]
If I know I am adding 10 images to the viewer, I want to listen to 10 `add-item` events (to correctly position them with #888) before unsubscribing.

**Comment by iangilman** [comment]
Fair enough. I guess I'm in favor of adding it as a fourth parameter to addOnceHandler. I know that's stretching the meaning of "once" but logically I think it fits in with the limited nature of that method.

**Comment by avandecreme** [comment]
Fourth parameter added :)

**Comment by iangilman** [comment]
Beautiful!

**Pull request merged**